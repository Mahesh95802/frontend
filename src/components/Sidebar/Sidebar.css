.sidebar{
    /* position: fixed; */
    overflow: scroll;
    top: 0;
    left: 0;
    width: 30vw;
    max-width: 300px;
    min-width: 170px;
    height: 100vh;
    background-color: rgb(37, 37, 37);
}

.sidebar-header{
    display: flex;
    /* flex: 2; */
    /* justify-content: center; */
    /* height: 50px; */
    max-height: 69px;
    align-items: center;
    font-weight: bolder;
    color: white;
    font-size: 25px;
    padding: 20px;
    background-color: blueviolet;
}

.sidebar h4{
    padding-top: 5px;
    padding-bottom: 5px;
    color: rgb(123, 123, 123);
}

.content-type{
    color: rgb(86, 86, 86);
    padding-top: 5px;
    padding-bottom: 5px;
    font-size: 15px;
}