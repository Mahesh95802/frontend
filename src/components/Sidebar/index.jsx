import React from 'react';
import PropTypes from 'prop-types';

import './Sidebar.css';

const Sidebar = ({ selected, contentTypes, navigate }) => {

	return (
		<div className="sidebar">
			<div className="sidebar-header">CMS+</div>
			<div className="collection-types">
				<div className="collection-types-header" >
					<h4 className="basic-padding">
                        COLLECTION TYPES
					</h4>
				</div>
				{
					contentTypes 
					&&  contentTypes.map((contentType) => {
						return (
							<div className="content-type" key={contentType.id} style={ selected === contentType.id ? { backgroundColor: 'black', color: 'white', fontWeight: 'bolder' } : null }  onClick={() => navigate(`/content-type/${contentType.id}`)}>
								<li className="basic-padding">{contentType.name}</li>
							</div>
							// <a className="content-type" href={`content-type/${contentType.id}`} key={contentType.id} style={ selected === contentType.id ? { backgroundColor: 'black', color: 'white', fontWeight: 'bolder' } : null } >
							// 	<li className="basic-padding">{contentType.name}</li>
							// </a>
						);
					})
				}
			</div>
			<div style={ selected ? null : { backgroundColor: 'black' } } className="content-type-builder">
				<h4 className="basic-padding" onClick={() => navigate('/home')} style={ selected ? null : { color: 'white' } }>
                    CONTENT TYPE BUILDER
				</h4>
			</div>

		</div>
	);
};

Sidebar.propTypes = {
	selected: PropTypes.number,
	contentTypes: PropTypes.array,
	navigate: PropTypes.func
};

export default Sidebar;