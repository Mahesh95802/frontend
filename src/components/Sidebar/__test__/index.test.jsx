/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import Sidebar from '..';

describe('Sidebar', () => {
    it('should render correctly', () => {
        const mockContentTypes = [
            {
                'id': 8,
                'name': 'Hello',
                'ContentSchemas': [
                    {
                        'id': 19,
                        'updatedAt': '2023-03-10T04:12:03.016Z'
                    }
                ],
                'Collections': []
            }
        ];
        const { asFragment } = render(<Sidebar contentTypes={mockContentTypes}/>);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should render correctly with selected', () => {
        const mockContentTypes = [
            {
                'id': 8,
                'name': 'Hello',
                'ContentSchemas': [
                    {
                        'id': 19,
                        'updatedAt': '2023-03-10T04:12:03.016Z'
                    }
                ],
                'Collections': []
            }
        ];
        const { asFragment } = render(<Sidebar contentTypes={mockContentTypes} selected={8}/>);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should navigate to home when click on "CONTENT TYPE BUILDER"', () => {
        const mockContentTypes = [
            {
                'id': 8,
                'name': 'Hello',
                'ContentSchemas': [
                    {
                        'id': 19,
                        'updatedAt': '2023-03-10T04:12:03.016Z'
                    }
                ],
                'Collections': []
            }
        ];
        const mockNavigate = jest.fn();
        const { getByText } = render(<Sidebar contentTypes={mockContentTypes} navigate={mockNavigate}/>);
        expect(mockNavigate).not.toHaveBeenCalled();
        getByText('CONTENT TYPE BUILDER').click();
        expect(mockNavigate).toHaveBeenCalledWith('/home');
    });
    it('should navigate to content type when click on "CONTENT TYPE"', () => {
        const mockContentTypes = [
            {
                'id': 8,
                'name': 'Hello',
                'ContentSchemas': [
                    {
                        'id': 19,
                        'updatedAt': '2023-03-10T04:12:03.016Z'
                    }
                ],
                'Collections': []
            }
        ];
        const mockNavigate = jest.fn();
        const { getByText } = render(<Sidebar contentTypes={mockContentTypes} navigate={mockNavigate}/>);
        expect(mockNavigate).not.toHaveBeenCalled();
        getByText('Hello').click();
        expect(mockNavigate).toHaveBeenCalledWith('/content-type/8');
    });
});