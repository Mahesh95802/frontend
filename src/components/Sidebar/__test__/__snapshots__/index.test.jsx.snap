// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sidebar"
  >
    <div
      class="sidebar-header"
    >
      CMS+
    </div>
    <div
      class="collection-types"
    >
      <div
        class="collection-types-header"
      >
        <h4
          class="basic-padding"
        >
          COLLECTION TYPES
        </h4>
      </div>
      <div
        class="content-type"
      >
        <li
          class="basic-padding"
        >
          Hello
        </li>
      </div>
    </div>
    <div
      class="content-type-builder"
      style="background-color: black;"
    >
      <h4
        class="basic-padding"
        style="color: white;"
      >
        CONTENT TYPE BUILDER
      </h4>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Sidebar should render correctly with selected 1`] = `
<DocumentFragment>
  <div
    class="sidebar"
  >
    <div
      class="sidebar-header"
    >
      CMS+
    </div>
    <div
      class="collection-types"
    >
      <div
        class="collection-types-header"
      >
        <h4
          class="basic-padding"
        >
          COLLECTION TYPES
        </h4>
      </div>
      <div
        class="content-type"
        style="background-color: black; color: white; font-weight: bolder;"
      >
        <li
          class="basic-padding"
        >
          Hello
        </li>
      </div>
    </div>
    <div
      class="content-type-builder"
    >
      <h4
        class="basic-padding"
      >
        CONTENT TYPE BUILDER
      </h4>
    </div>
  </div>
</DocumentFragment>
`;
